sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -313,6 +313,25 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     comparePlans(optimized, expected)
   }
 
+  test("SPARK-38531: Nested field pruning for Project and PosExplode") {
```

Comment:
I think the test in `ColumnPruningSuite` is sufficient. We didn't change `NestedColumnAliasing` object in this PR.